### DISCUSSION TITLE
How to deploy fastapi on windows server 2019?

### DISCUSSION THREAD

**Discussion by ftqiao** [fastapi/fastapi#8863] [Questions]
### Description
 I want to depoly app to windows server . I use 'python main.py ' in cmd line to run my application ,and I don't think it the right way.

**Comment by badger-py** [comment]
Hi!
Buy Linux server :)
On the Linux server install nginx, setup it as a proxy server. Run uvicorn is systemd unit

**Comment by maxiaojunwolf** [comment]
# I use IIS,but there are a lot of problems, like the on_startup、on_shutdown  funcs do't work,win32com.......
from a2wsgi import ASGIMiddleware
wsgi_app = ASGIMiddleware(app)

**Comment by YuriiMotov** [comment]
See: https://fastapi.tiangolo.com/deployment/
